performance-tests/bench_2/test_controller/ScenarioManager.cpp

```diff
@@ -201,107 +202,254 @@ void ScenarioManager::customize_configs(std::map<std::string, std::string>& work
   }
 }
 
-AllocatedScenario ScenarioManager::allocate_scenario(
-  const ScenarioPrototype& scenario_prototype, Nodes& available_nodes, bool debug_alloc)
+bool ScenarioManager::is_matched(const std::string& str, const std::string& pat) const
+{
+  if (pat.empty() ||
+      (!OpenDDS::DCPS::is_wildcard(pat.c_str()) && pat == str) ||
+      (OpenDDS::DCPS::is_wildcard(pat.c_str()) && ACE::wild_match(str.c_str(), pat.c_str(), true, true))) {
+    return true;
+  }
+  return false;
+}
+
+AllocatedScenario ScenarioManager::allocate_scenario(const ScenarioPrototype& scenario_prototype,
```

Comment:
I think this is complicated enough that it needs a unit test. For now, let's just make a test app local to the test_controller directory. Feel free to use googletest and make it a required library, but you'll need to update the mpc accordingly. Some of the security tests use googletest already, if you're looking for examples of how we do that.

We should test for a few basic scenarios, including failures, and then maybe one or two more complicated scenarios.

Reply:
If it makes sense to separate the choice of node_controller for each worker config out from the reading in of worker config files that happens in this function in order to make testing easier, I'm fine with that.